activerecord/test/cases/connection_adapters/merge_and_resolve_default_url_config_test.rb

```diff
@@ -124,6 +124,17 @@ def test_resolver_with_database_uri_and_supplied_url
         assert_equal expected, actual.configuration_hash
       end
 
+      def test_resolver_with_database_uri_containing_only_database_name
+        ENV["DATABASE_URL"] = "foo"
+        ENV["RAILS_ENV"] = "test"
```

Comment:
This might leakin other tests. I think we got some `with_env` helper to set env and restore it after.

Reply:
It is restored back in the `teardown` - https://github.com/rails/rails/pull/53628/files#diff-1db0a36e5ef2eb70098c1f18dc8d9b8bc39d3619d83ce1e8456172354bb120daR19

Reply:
Ah my bad.